### DISCUSSION TITLE
Safari doesn't work

### DISCUSSION THREAD

**Discussion by Kappo99** [tailwindlabs/tailwindcss#12463] [Help] [answered]
I created a website with tailwindcss and html.
On Windows with Chrome, Firefox, Edge it works correctly.
On MacOS with Chrome is everything ok, but with Safari there are so many layout troubles.
I checked the webkit clauses and they are good. What can be the problem?
How can I do some test to find the problem?

**Comment by wongjn** [comment]
> What can be the problem?

It could be an old version of Safari that does not support some of the more modern CSS features that Tailwind uses.

> How can I do some test to find the problem?

You can check the problem HTML elements in the developer tools of the browser and see if any properties are not recognized/marked as invalid.

**Reply by wongjn** [reply]
> So Safari 16.4 is too old?[^1]

Nope! That should be within the support range of Tailwind v3.

> In HTML elements I cannot see the related CSS of the navbar elements, so these elements shows up with no CSS[^2]

That could be the problem – for some reason, some CSS isn't being applied to those elements properly.

[^1]: https://github.com/tailwindlabs/tailwindcss/discussions/12463#discussioncomment-7635691
[^2]: https://github.com/tailwindlabs/tailwindcss/discussions/12463#discussioncomment-7635700

**Reply by Kappo99** [reply]
Are there some tips to find the problem?
Why some CSS is not being applied correctly...

**Reply by Kappo99** [reply]
With Safari HTML Elements, in tab Sources I see some CSS class written on red color. How can I know what is the problem? Because they are the class that are not apply correctly to the elements

**Reply by Kappo99** [reply]
I think the problem is nested CSS elements.
For example
`.footer {
    @apply ...;
    .logo {
        @apply ...;
    }
}`
Everything in .footer works correctly, but .logo and all others nested elements are written in red and doesn't work on Safari.

Is there a way to have nested elements on Safari or I have to change all my CSS codes?

**Reply by Kappo99** [reply, marked as answer]
I resolved the issue. It was my mistake.
I wrote with nested CSS elements, but I didn't run a compiler like postcss, so the nested elements are not supported with all browsers (e.g. Safari 16.4 didn't work, but Safari updated on 17.1 works).
Now the postcss converts nested elements into standard CSS elements, so it works on all browsers.

Thanks for the previus suggestions

**Comment by Kappo99** [comment]
So Safari 16.4 is too old? This is a big problem, if a 6 months ago version is too old.
Most devices in use by users cannot see my website correctly

**Reply by wongjn** [reply]
Replied at https://github.com/tailwindlabs/tailwindcss/discussions/12463#discussioncomment-7635717 to keep the reply chain together.

**Comment by Kappo99** [comment]
In HTML elements I cannot see the related CSS of the navbar elements, so these elements shows up with no CSS

**Reply by wongjn** [reply]
Replied at https://github.com/tailwindlabs/tailwindcss/discussions/12463#discussioncomment-7635717 to keep the reply chain together.
